src/components/Dialog/__tests__/Dialog.interactions.ts

```diff
@@ -0,0 +1,59 @@
+import { resetFocus } from "../../../__tests__/interactions-helper";
+import { expect } from "@storybook/jest";
+import { BoundFunctions, Screen } from "@testing-library/react";
+import {
+  CONTEXT_MENU_DIALOG,
+  CLICK_OUTSIDE_DIALOG,
+  HIDE_TRIGGERS_CONTAINER,
+  CLICK_OUTSIDE_DIALOG_BUTTON
+} from "./DialogDataTestIds";
+import { getByTestId, interactionSuite } from "../../../tests/interactions-utils";
+import { queries, userEvent, fireEvent, waitFor } from "@storybook/testing-library";
+
+const isDialogHiddenAfterClickOutside = createTestIfDialogHiddenAfterTrigger(CLICK_OUTSIDE_DIALOG, () =>
+  userEvent.click(getDialogContainer())
+);
+const isDialogHiddenAfterContextMenu = createTestIfDialogHiddenAfterTrigger(CONTEXT_MENU_DIALOG, () =>
+  fireEvent.contextMenu(getDialogContainer())
+);
+
+export const closeTriggersInteractionSuite = interactionSuite({
+  tests: [isDialogHiddenAfterClickOutside, isDialogHiddenAfterContextMenu],
+  beforeAll: async canvas => {
+    // From some reason we have an issue with rendering the dialogs according to the container selector in the initial mount, after clicking
+    // all dialog render in the right placements
+    const clickOutsideButton = await getByTestId(canvas, CLICK_OUTSIDE_DIALOG_BUTTON);
+    await userEvent.click(clickOutsideButton);
+    // wait for hide default delay
+    await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
instead of statically waiting - I suggest a retry mechanism that waits until the expected result happens. in this case - the modal should open

Looks at something like this or use a library that does it
https://learnersbucket.com/examples/interview/retry-promises-n-number-of-times-in-javascript/
https://github.com/wix-incubator/wix-eventually

Reply:
Used storybook waitFor ability which provide similar mechanism